presto-main/src/main/java/com/facebook/presto/sql/planner/RelationPlanner.java

```diff
@@ -299,6 +301,23 @@ else if (firstDependencies.stream().allMatch(right.canResolvePredicate()) && sec
                 Optional.empty(),
                 Optional.empty());
 
+        if (node.getType() != INNER) {
+            List<InPredicate> inPredicateSubqueries = analysis.getInPredicateSubqueries(node);
+            for (Expression complexExpression : complexJoinExpressions) {
+                for (InPredicate inPredicate :  inPredicateSubqueries) {
+                   if (nodeContains(complexExpression, inPredicate)) {
+                       throw new SemanticException(
+                               NOT_SUPPORTED,
+                               inPredicate,
+                               "IN with subquery predicate in join condition not supported");
+                   }
+                }
+            }
+
+            // non inner join subqueries can be applied only to one side of join
+            leftPlanBuilder = subqueryPlanner.handleSubqueries(leftPlanBuilder, complexJoinExpressions, node);
```

Comment:
What's special about the left side? Is this only for left outer joins? What happens for full or right outer joins?